nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/elasticsearch/ElasticsearchObject.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.rest.management.elasticsearch;
+
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.automation.elasticsearch.ElasticsearchIndexOperation;
+import org.nuxeo.ecm.core.api.CloseableCoreSession;
+import org.nuxeo.ecm.core.api.CoreInstance;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.model.Repository;
+import org.nuxeo.ecm.core.repository.RepositoryService;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+import org.nuxeo.elasticsearch.api.ElasticSearchAdmin;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * @since 11.1
+ */
+@WebObject(type = "elasticsearch")
+public class ElasticsearchObject extends DefaultObject {
+
+    /**
+     * Runs an ES indexation on the given repository.
+     *
+     * @param repositoryName the repository name for which we want to start the ES indexation
+     * @throws NuxeoException
+     */
+    @POST
+    @Path("{repositoryName}/reindex")
+    public void doIndexing(@PathParam("repositoryName") String repositoryName) {
+        Repository repository = getRepository(repositoryName);
+        Framework.doPrivileged(() -> {
+            try (CloseableCoreSession coreSession = CoreInstance.openCoreSessionSystem(repository.getName())) {
+                AutomationService automationService = Framework.getService(AutomationService.class);
+                automationService.run(new OperationContext(coreSession), ElasticsearchIndexOperation.ID);
```

Comment:
this was my first idea before starting the whole development, after a short discussion with @kevinleturc he advice me to use the Operation and i am going to use the Bulk indexing operation as proposed by @ataillefer instead of the regular one :)